src/main/scala/de/upb/cs/swt/delphi/instanceregistry/daos/DatabaseInstanceDAO.scala

```diff
@@ -0,0 +1,544 @@
+
+    if(hasData)
+    {
+      val resultAll = Await.result(db.run(instances.filter(_.componentType === cmpType).result), Duration.Inf)
+      val listAll = List() ++ resultAll.map(_.value)
+      listInstance = listAll.map(c => dataToObjectInstance(Option(c)))
+    }
+    listInstance
+  }
+
+  override def allInstances() : List[Instance] = {
+    var listInstance = List[Instance]()
+    val hasData = Await.result(db.run(instances.exists.result), Duration.Inf)
+    if(hasData){
+      val resultAll = Await.result(db.run(instances.result), Duration.Inf)
+      val listAll = List() ++ resultAll.map(_.value)
+      listInstance = listAll.map(c => dataToObjectInstance(Option(c)))
+    }
+
+    listInstance
+  }
+
+  override def removeAll() : Unit = {
+    removeAllInstances()
+    removeAllInstanceMatchingResults()
+    removeAllInstanceEvents()
+    removeAllInstanceLinks()
+    dumpToRecoveryFile()
+  }
+
+  override def addMatchingResult(id: Long, matchingSuccessful : Boolean) : Try[Unit] = {
+    if(hasInstance(id)){
+      val addAction: DBIO[Unit] = DBIO.seq(
+        instanceMatchingResults.map(c => (c.instanceId, c.matchingSuccessful))
+                                += (id, matchingSuccessful)
+      )
+      db.run(addAction)
+      Success(log.info(s"Successfully added matching result $matchingSuccessful to instance with id $id."))
+    } else {
+      log.warning(s"Cannot add matching result, instance with id $id not present.")
+      Failure(new RuntimeException(s"Cannot add matching result, instance with id $id not present."))
+    }
+  }
+
+
+  override def getMatchingResultsFor(id: Long) : Try[List[Boolean]] = {
+    if(hasInstance(id) && hasMatchingResultForInstance(id)){
+      val query = instanceMatchingResults.filter(_.instanceId === id).map(_.matchingSuccessful).result
+      val result = Await.result(db.run(query), Duration.Inf).toList
+      Success(result)
+    } else {
+      log.warning(s"Cannot get matching results, id $id not present!")
+      Failure(new RuntimeException(s"Cannot get matching results, id $id not present!"))
+    }
+  }
+
+  override def initialize() : Unit = {
+
+    if(dbTest()){
+        val tables = List(instances, instanceEvents, instanceLinks, instanceMatchingResults)
+        val existing = db.run(MTable.getTables)
+        val createAction = existing.flatMap( v => {
+          val names = v.map(mt => mt.name.name)
+          val createIfNotExist = tables.filter( table =>
+            !names.contains(table.baseTableRow.tableName)).map(_.schema.create)
+          db.run(DBIO.sequence(createIfNotExist))
+        })
+        Await.result(createAction, Duration.Inf)
+
+
+      log.info("Initializing dynamic instance DAO...")
+      clearData()
+      log.info("Successfully initialized.")
+    } else {
+      log.error("Not found any database with the provided settings.")
+
+      val terminationFuture = system.terminate()
+
+      terminationFuture.onComplete {
+        sys.exit(0)
```

Comment:
Please do not terminate the system and do not call ```sys.exit``` here, it will terminate the whole sbt process on the CI servers. Maybe you could modify your dbTest() method to throw the exception it found instead of just returning false ? This way the exception would be thrown all the way to the main method and the registry would terminate without terminating the whole testing / build process.